core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1747,6 +1748,12 @@ object SnappySession extends Logging {
   private[this] val ID = new AtomicInteger(0)
   private[sql] val ExecutionKey = "EXECUTION"
 
+  lazy val isEnterpriseEdition: Boolean = {
+    GemFireVersion.getInstance(Class.forName("io.snappydata.gemxd.SnappyDataVersion"),
```

Comment:
This is the same as that added in SnappyDataVersion.isEnterpriseEdition. Please make that as lazy val and use it removing this.